flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraph.java

```diff
@@ -266,7 +268,10 @@ public boolean isIterative() {
 			String operatorName) {
 
 		if (operatorFactory.isStreamSource()) {
-			addNode(vertexID, slotSharingGroup, coLocationGroup, SourceStreamTask.class, operatorFactory, operatorName);
+			Class<? extends AbstractInvokable> invokableClass =
```

Comment:
This looks fragile to me. See the top-level comment about an idea for a cleaner way of handling this.